flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/iomanager/IOManager.java

```diff
@@ -224,6 +238,8 @@ public void deleteChannel(FileIOChannel.ID channel) throws IOException {
 
 	public abstract BufferFileReader createBufferFileReader(FileIOChannel.ID channelID, RequestDoneCallback<Buffer> callback) throws IOException;
 
+	public abstract BufferFileReader createBufferOrEventFileReader(FileIOChannel.ID channelID, RequestDoneCallback<Buffer> callback) throws IOException;
```

Comment:
If we go forward, I would suggest to factor this out into an interface (that is also implemented by `IOManager`) so we do not pass such a big dependency. This component does not need to know about the IOManager as a whole if it is only interested in this sub-functionality.